src/Compilers/CSharp/Test/Semantic/Utilities/ValueSetTests.cs

```diff
@@ -0,0 +1,663 @@
+                int i2 = Random.Next(int.MinValue, int.MaxValue);
+                if (i1 > i2) (i1, i2) = (i2, i1);
+                if ((i1 + 1) >= i2) continue;
+                IValueSet<int> values1 = ForInt.Related(LessThanOrEqual, i1).Union(ForInt.Related(GreaterThanOrEqual, i2));
+                Assert.Equal($"[{int.MinValue}..{i1}],[{i2}..{int.MaxValue}]", values1.ToString());
+                IValueSet<int> values2 = ForInt.Related(GreaterThanOrEqual, i2).Union(ForInt.Related(LessThanOrEqual, i1));
+                Assert.Equal(values1, values2);
+            }
+        }
+
+        [Fact]
+        public void TestUnion_02()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue + 1, int.MaxValue);
+                int i2 = Random.Next(int.MinValue, int.MaxValue);
+                if (i1 < i2) (i1, i2) = (i2, i1);
+                IValueSet<int> values1 = ForInt.Related(LessThanOrEqual, i1).Union(ForInt.Related(GreaterThanOrEqual, i2));
+                Assert.Equal($"[{int.MinValue}..{int.MaxValue}]", values1.ToString());
+                IValueSet<int> values2 = ForInt.Related(GreaterThanOrEqual, i2).Union(ForInt.Related(LessThanOrEqual, i1));
+                Assert.Equal(values1, values2);
+            }
+        }
+
+        [Fact]
+        public void TestComplement_01()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue + 1, int.MaxValue);
+                int i2 = Random.Next(int.MinValue, int.MaxValue);
+                if (i1 > i2) (i1, i2) = (i2, i1);
+                if ((i1 + 1) >= i2) continue;
+                IValueSet<int> values1 = ForInt.Related(LessThanOrEqual, i1).Union(ForInt.Related(GreaterThanOrEqual, i2));
+                Assert.Equal($"[{int.MinValue}..{i1}],[{i2}..{int.MaxValue}]", values1.ToString());
+                IValueSet<int> values2 = values1.Complement();
+                Assert.Equal($"[{i1 + 1}..{i2 - 1}]", values2.ToString());
+            }
+        }
+
+        [Fact]
+        public void TestAny_01()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue, int.MaxValue);
+                int i2 = Random.Next(int.MinValue, int.MaxValue);
+                if (i1 > i2) (i1, i2) = (i2, i1);
+                IValueSet<int> values = ForInt.Related(GreaterThanOrEqual, i1).Intersect(ForInt.Related(LessThanOrEqual, i2));
+                Assert.Equal($"[{i1}..{i2}]", values.ToString());
+                test(int.MinValue);
+                if (i1 != int.MinValue) test(i1 - 1);
+                test(i1);
+                test(i1 + 1);
+                test(int.MaxValue);
+                if (i2 != int.MinValue) test(i2 - 1);
+                test(i2);
+                test(i2 + 1);
+                test(Random.Next(int.MinValue, int.MaxValue));
+                test(Random.Next(int.MinValue, int.MaxValue));
```

Comment:
>test(Random.Next(int.MinValue, int.MaxValue)); [](start = 16, length = 46)

Duplicate test.

Reply:
Will address in next PR.

---
In reply to: [383427112](https://github.com/dotnet/roslyn/pull/41503#discussion_r383427112) [](ancestors = 383427112)